contracts/Strategy.sol

```diff
@@ -115,14 +137,17 @@ contract Strategy is BaseStrategy {
     }
 
     function estimatedTotalAssets() public view override returns (uint256) {
-        return
+        uint256 totalAssets =
             balanceOfWant().add(estimatedProfit()).add(
-                sUSDToWant(
-                    balanceOfSusdInVault().add(balanceOfSusd()).sub(
-                        balanceOfDebt()
-                    )
-                )
+                sUSDToWant(balanceOfSusdInVault().add(balanceOfSusd()))
             );
+        uint256 totalLiabilities = sUSDToWant(balanceOfDebt());
+        // NOTE: the ternary operator is required because debt can be higher than assets
+        // due to i) increase in debt or ii) losses in invested assets
+        return
+            totalAssets > totalLiabilities
```

Comment:
i understand the idea behind returning 0 when liability is higher. But it gives me some alarm in code like the below were we had an active position and it went to 0 the strategy is no longer consider active, altho i can think of a critical consequence.

https://github.com/yearn/yearn-vaults/blob/f97088d1313044e43fe2f8c6e98641d10544abd4/contracts/BaseStrategy.sol#L519

Reply:
I guess if this condition where liabilities become higher than total assets it means that something went very wrong with the strategy management under normal circumstances?

Reply:
interesting point

this is super low probability (almost negligible) because it would mean that the debt in want terms is higher than the full collat + cash borrowed in want terms. 

taking into account that Synthetix requires a 500% c-ratio (5 times collat vs debt) and liquidates positions at 200% c-ratio (2 times collat vs debt), this would mean that:
- debt pool value has +5x'ed between harvests (which would mean that btc / eth has gone up 500% in less than a week)
- SNX price has collapsed 80%+

and that the strategy has not been liquidated in time by the protocol.

we are in serious losses once the debt pool value has gone up +150% between harvests (and the SNX price has remained stable). this itself "has never happened"  according to Synthetix core devs.

Reply:
missed your second comment. 

the TL;DR from my previous answer is: things have to go very very wrong for liabilities to be higher than assets.